modules/common/condition/funcs_test.go

```diff
@@ -165,6 +165,55 @@ func TestSet(t *testing.T) {
 	g.Expect(c2.LastTransitionTime).To(BeIdenticalTo(time1))
 }
 
+func TestRemove(t *testing.T) {
+	conditions := Conditions{}
+
+	tests := []struct {
+		name      string
+		condition *Condition
+		want      Conditions
+	}{
+		{
+			name:      "Add nil condition",
+			condition: nil,
+			want:      CreateList(unknownReady),
+		},
+		{
+			name:      "Add unknownB condition",
+			condition: unknownB,
+			want:      CreateList(unknownReady, unknownB),
+		},
+		{
+			name:      "Add another condition unknownA, gets sorted",
```

Comment:
these names feels like copy paste from above. I suggest to align them to the current test case.

Reply:
Align? You want the initial conditions slice to be created differently?

Reply:
The L171 - L196 code is testing what happens when you add different conditions to the list one by one. Asserting things like sorting behavior of the list. (Hence the name of here says "gets sorted").  But this logic is already tested by other test cases above. So please don't repeat it.

What you need is a condition list to remove conditions from. That can be done as simply as
```golang
	conditions := CreateList(unknownReady, unknownA, unknownB)
```
Then you can do L198-... to test the removals as you proposed.

Reply:
I'm sure this can be optimized. Not sure what's the harm in having a few lines of extra test code to create the list like in other tests. I can change if you insist though.

Reply:
This is not optimization this is removal of duplicated test code. Less code means less logic to understand and maintain.

Reply:
By optimization I did not mean only code optimization for performance, but also 'test' code readability. Anyway, now that I understand the ginkgo/gomega frameworks better, updated the tests. Can we now move on with it to unblock the openstack-operator patch?.